toml/src/main/kotlin/org/rust/toml/completion/CargoTomlFeatureDependencyCompletionProvider.kt

```diff
@@ -32,7 +34,10 @@ class CargoTomlFeatureDependencyCompletionProvider : CompletionProvider<Completi
     override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
         val element = parameters.position
         val tomlFile = element.containingFile as? TomlFile ?: return
+
+        val parentFeature = element.parentOfType<TomlKeyValueImpl>()?.key ?: return
         for (feature in tomlFile.allFeatures()) {
+            if (feature == parentFeature) continue
```

Comment:
Features depend on each other like a tree, so just checking a parent is not enough. Let's build a feature graph and check cycles based on it

Reply:
IIUC Cargo features can be cyclic (https://github.com/rust-lang/cargo/pull/4473), but only optional dependencies, as well as related features, cannot be